OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/SensitiveWorkshopsServiceTests.cs

```diff
@@ -0,0 +1,227 @@
+                ministryAdminServiceMock.Object,
+                regionAdminServiceMock.Object,
+                codeficatorServiceMock.Object);
+
+        userId = Guid.NewGuid().ToString();
+    }
+
+    #region FetchByFilterForAdmins
+
+    [Test]
+    public async Task FetchByFilterForAdmins_RoleRegionAdmin_ShouldBuildPredicateAndReturnEntities()
+    {
+        // Arrange
+        var institutionId = Guid.NewGuid();
+        var parentCATOTTGId = 1;
+        WorkshopFilterAdministration filter = new WorkshopFilterAdministration();
+        var admin = new RegionAdminDto() { Id = userId, InstitutionId = institutionId, CATOTTGId = parentCATOTTGId };
+        var resultExcpected = SetupFetchByFilterForAdmins(userId, true, false, parentCATOTTGId, filter, admin);
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins().ConfigureAwait(false);
+
+        // Assert
+        resultExcpected.Should().BeEquivalentTo(result);
+    }
+
+    [Test]
+    public void FetchByFilterForAdmins_WhenFilterIsNullRegionAdminIsNull_ShouldReturnException()
+    {
+        // Arrange
+        var parentCATOTTGId = 1;
+        RegionAdminDto admin = null;
+        SetupFetchByFilterForAdmins(userId,  true, false, parentCATOTTGId, null, admin);
+
+        // Assert
+        Func<Task<SearchResult<WorkshopDto>>> result = () => sensitiveWorkshopService.FetchByFilterForAdmins();
+
+        // Act
+        result.Should().ThrowAsync<InvalidOperationException>()
+        .WithMessage($"Region admin with the specified ID: {userId} not found");
+    }
+
+    [Test]
+    public async Task FetchByFilterForAdmins_WhenFilterByCATOTTGId_ShouldReturnEntities()
+    {
+        // Arrange
+        var parentCATOTTGId = 10;
+        WorkshopFilterAdministration filter = new WorkshopFilterAdministration() { From = 3, Size = 10, CATOTTGId = 1 };
+        var resultExcpected = SetupFetchByFilterForAdmins(userId, false, false, parentCATOTTGId, filter);
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins(filter).ConfigureAwait(false);
+
+        // Assert
+        resultExcpected.Should().BeEquivalentTo(result);
+    }
+
+    [Test]
+    public async Task FetchByFilterForAdmins_WhenRoleMinistryAdmin_ShouldReturnEntities()
+    {
+        // Arrange
+        var institutionId = Guid.NewGuid();
+        var parentCATOTTGId = 1;
+        var admin = new MinistryAdminDto() { Id = userId, InstitutionId = institutionId };
+        WorkshopFilterAdministration filter = new WorkshopFilterAdministration() { Size = 8, CATOTTGId = 1, From = 2, SearchString = "key word" };
+        var resultExcpected = SetupFetchByFilterForAdmins(userId, false, true, parentCATOTTGId, filter, null, admin);
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins(filter).ConfigureAwait(false);
+
+        // Assert
+        resultExcpected.Should().BeEquivalentTo(result);
+    }
+
```

Comment:
Same conmment about testing keywords filtering separately and in combination with other filters

Reply:
Added various cases for search